geometric_kernels/spaces/homogeneous_spaces.py

```diff
@@ -0,0 +1,184 @@
+import abc
+
+import lab as B
+import numpy as np
+
+from geometric_kernels.spaces.base import DiscreteSpectrumSpace
+from geometric_kernels.spaces.eigenfunctions import (
+    Eigenfunctions,
+    EigenfunctionWithAdditionTheorem,
+)
+from geometric_kernels.spaces.lie_groups import LieGroupCharacter, MatrixLieGroup
+
+
+class CompactHomogeneousSpaceAddtitionTheorem(EigenfunctionWithAdditionTheorem):
```

Comment:
See the comment about class naming for the respective class in lie_groups.py. Also, we don't necessarily need to average characters to get zonal spherical functions for homogeneous spaces, sphere is an example. We could reflect this in naming, like CompactHomogeneousSpaceAveragingEigenfunctions or CompactHomogeneousSpaceAveragingAddtitionTheorem. Though this is sort of ugly

Reply:
\+ docstrings everywhere

Reply:
Not sure about "the comment about class naming for the respective class in lie_groups.py". This seems to be already addressed? `CompactHomogeneousSpaceAveragingAddtitionTheorem` is too verbose.